### ISSUE TITLE
400 bad request when trying to cancel/edit large order

### ISSUE THREAD

**Issue by danFbach** [nopSolutions/nopCommerce#6723]
nopCommerce version: 4.60.3

Steps to reproduce the problem: 

The order that I'm working with has 432 items.

When i try to cancel the order with the `cancel` button, or perform a manual status change, the change fails and the user receives a blank page. Dev tools and nop log show 400 bad request. 

Since the edit page is one big form that posts all input values regardless of what action is being taken on the order, and each order item has 6 attributes being posted, this results in 2,592 form fields being posted + approximately 25 other fields. So, the edit form for this order is submitting over 2,600 fields. This exceeds the default value of 1024 for the `RequestFormLimits` action attribute. I ran into a similar problem when developing a plugin for wholesale ordering.

The best fix, in my opinion would involve limiting the form fields to only submit the values to what is necessary for what is being changed.

The quick, less secure/robust fix would be adding `[DisableRequestSizeLimit]`, `[RequestFormLimits(ValueCountLimit = int.MaxValue, KeyLengthLimit = int.MaxValue)]` attributes to the `CancelOrder` action and other edit related actions which could suffer from this same issue.

I can certainly apply the attribute overrides to my local build to get it working, but I would like to see a more robust upstream fix for this.

**Comment by danFbach** [comment]
I was unable to resolve this by adding the action attributes. 
continuing to investigate.
there is no issue canceling smaller, normal sized, orders.

**Comment by AndreiMaz** [comment]
also see https://github.com/nopSolutions/nopCommerce/issues/5950

**Comment by danFbach** [comment]
Okay, I have a solution, but it should probably be assessed for its potential for abuse, so I don't this it's appropriate submit a PR until it's been discussed. 
Also...
```cs
services.Configure<FormOptions>(options =>
{
    options.ValueCountLimit = int.MaxValue;
});
```
could be added in various places - i just found this to be the most appropriate.

Source: [https://stackoverflow.com/questions/38357108/form-submit-resulting-in-invaliddataexception-form-value-count-limit-1024-exce](url)

In `src\Presentation\Nop.Web.Framework\Infrastructure\Extensions\ServiceCollectionExtensions.cs`

```cs
public static void ConfigureApplicationServices(this IServiceCollection services,
            WebApplicationBuilder builder)
        {
            //add accessor to HttpContext
            services.AddHttpContextAccessor();

            //initialize plugins
            var mvcCoreBuilder = services.AddMvcCore();
            var pluginConfig = new PluginConfig();
            builder.Configuration.GetSection(nameof(PluginConfig)).Bind(pluginConfig, options => options.BindNonPublicProperties = true);
            mvcCoreBuilder.PartManager.InitializePlugins(pluginConfig);

            //allow large form submissions
            services.Configure<FormOptions>(options =>
            {
                options.ValueCountLimit = int.MaxValue;
            });

            //create engine and configure service provider
            var engine = EngineContext.Create();

            engine.ConfigureServices(services, builder.Configuration);
        }

```

**Comment by skoshelev** [comment]
Closed #6723

**Comment by jigarsangoi** [comment]
Hello everyone,
I come across issue where if you open multiple tabs then only last opened tab will work without issue. when you try to perform any action in previously opened tabs will generate Error 400 bad Request. This problem coming in 4.60 not sure about older version. So can you suggest to avoid this ? Also please check this link [https://stackoverflow.com/questions/78061845/asp-net-core-antiforgery-token-that-works-over-multiple-tabs](url) @RomanovM  @skoshelev
